src/iidm/converter/xml/StaticVarCompensatorXml.cpp

```diff
@@ -37,8 +37,14 @@ const char* StaticVarCompensatorXml::getRootElementName() const {
 StaticVarCompensator& StaticVarCompensatorXml::readRootElementAttributes(StaticVarCompensatorAdder& adder, NetworkXmlReaderContext& context) const {
     const auto& bMin = context.getReader().getAttributeValue<double>(B_MIN);
     const auto& bMax = context.getReader().getAttributeValue<double>(B_MAX);
-    double voltageSetpoint = context.getReader().getOptionalAttributeValue(VOLTAGE_SET_POINT, stdcxx::nan());
-    double reactivePowerSetpoint = context.getReader().getOptionalAttributeValue(REACTIVE_POWER_SET_POINT, stdcxx::nan());
+    std::string voltageSetpointName = VOLTAGE_SETPOINT;
+    std::string reactivePowerSetpointName = REACTIVE_POWER_SETPOINT;
+    IidmXmlUtil::runUntilMaximumVersion(IidmXmlVersion::V1_2(), context.getVersion(), [&voltageSetpointName, &reactivePowerSetpointName]() {
+        voltageSetpointName = VOLTAGE_SET_POINT;
+        reactivePowerSetpointName = REACTIVE_POWER_SET_POINT;
+    });
+    double voltageSetpoint = context.getReader().getOptionalAttributeValue(voltageSetpointName, stdcxx::nan());
+    double reactivePowerSetpoint = context.getReader().getOptionalAttributeValue(reactivePowerSetpointName, stdcxx::nan());
```

Comment:
honestly, I'm not sure this implementation is really readable even if it matches to what is done in the Java version. I would prefer to initialize each voltageSetpointName and reactivePowerSetpointName using a ternary operator or a classic if/else block

What do you think?

Reply:
Fully agree, I would prefer using a ternary operator.

Reply:
Fixed in d35a81e890dba299cbc45f2b41fcd95364b5979b